File: fe/src/api/system/permission.ts
import request from '@/utils/request';

export interface IPermission {
    id?: number;
    name: string;
    code: string;
    parent_id?: number;
    children_ids?: number[];
    children?: IPermission[];
    menu_ids?: number[];
    frontend_route_ids?: number[];
    backend_route_ids?: number[];
}

// 获取权限
export const getApi = (id: number): IRespData<Required<IPermission>> => request.get("/Permission/Get", { params: { id } })
// 创建权限
export const addApi = (data: IPermission): IRespData<null> => request.post("/Permission/Add", data)
// 编辑权限
export const editApi = (data: IPermission): IRespData<null> => request.post("/Permission/Edit", data)
// 查询所有顶级权限
export const listAllTopApi = (): IRespData<Required<IPermission>[]> => request.get("/Permission/ListAllTop")
// 查询权限树
export const listTreeApi = (): IRespData<Required<IPermission>[]> => request.get("/Permission/ListTree")
// 删除权限
export const deleteApi = (id: number): IRespData<null> => request.delete("/Permission/Delete", { params: { id } })

File: app/models/view/system/Permission.go
package v_system

import orm_system "github.com/zze326/devops-helper/app/models/orm/system"

type AddPermissionReq struct {
	Name             string `json:"name" valid:"required~权限名称不能为空"`
	Code             string `json:"code" valid:"required~权限代码不能为空"`
	ParentID         int    `json:"parent_id"`
	ChildrenIDs      []int  `json:"children_ids"`
	MenuIDs          []int  `json:"menu_ids"`
	FrontendRouteIDs []int  `json:"frontend_route_ids"`
	BackendRouteIDs  []int  `json:"backend_route_ids"`
}

type EditPermissionReq struct {
	ID int `json:"id" valid:"int~权限ID必须为整型数字,gt(0)~权限ID必须大于0"`
	AddPermissionReq
}

type GetPermissionResp struct {
	ID               int                      `json:"id"`
	Code             string                   `json:"code"`
	Name             string                   `json:"name"`
	ParentID         int                      `json:"parent_id"`
	Children         []*orm_system.Permission `json:"children"`
	ChildrenIDs      []int                    `json:"children_ids"`
	MenuIDs          []int                    `json:"menu_ids"`
	FrontendRouteIDs []int                    `json:"frontend_route_ids"`
	BackendRouteIDs  []int                    `json:"backend_route_ids"`
}
